examples/chip-tool/commands/discover/Commands.h

```diff
@@ -0,0 +1,78 @@
+/*
+ *   Copyright (c) 2021 Project CHIP Authors
+ *   All rights reserved.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ *
+ */
+
+#pragma once
+
+#include "Command.h"
+#include <mdns/Resolver.h>
+
+class Discover : public Command, public chip::Mdns::ResolverDelegate
+{
+public:
+    Discover() : Command("resolve-node-id") { AddArgument("nodeid", 0, UINT64_MAX, &mNodeId); }
+
+    CHIP_ERROR Run(PersistentStorage & storage, NodeId localId, NodeId remoteId) override
+    {
+        ReturnErrorOnFailure(mCommissioner.SetUdpListenPort(storage.GetListenPort()));
+        ReturnErrorOnFailure(mCommissioner.Init(localId, &storage));
+        ReturnErrorOnFailure(mCommissioner.ServiceEvents());
+
+        ReturnErrorOnFailure(chip::Mdns::Resolver::Instance().SetResolverDelegate(this));
+        ReturnErrorOnFailure(chip::Mdns::Resolver::Instance().ResolveNodeId(mNodeId, 0, chip::Inet::kIPAddressType_Any));
```

Comment:
fyi: embedded apps which use app/server/Mdns.cpp use test fabric id 5544332211:
```
    if (DeviceLayer::ConfigurationMgr().GetFabricId(fabricId) != CHIP_NO_ERROR)
    {
        ChipLogError(Discovery, "Fabric ID not known. Using a default");
        fabricId = 5544332211;
    }
```
Consider adding a parameter for the fabric ID until it's unified across platforms.